hardware_interface/src/resource_manager.cpp

```diff
@@ -334,6 +338,47 @@ size_t ResourceManager::system_components_size() const
   return resource_storage_->systems_.size();
 }
 
+std::unordered_map<std::string, status> ResourceManager::get_components_status()
+{
+  for (auto & component : resource_storage_->actuators_) {
+    resource_storage_->hardware_status_map_[component.get_name()] = component.get_status();
+  }
+  for (auto & component : resource_storage_->sensors_) {
+    resource_storage_->hardware_status_map_[component.get_name()] = component.get_status();
+  }
+  for (auto & component : resource_storage_->systems_) {
+    resource_storage_->hardware_status_map_[component.get_name()] = component.get_status();
+  }
+
+  return resource_storage_->hardware_status_map_;
+}
+
+void ResourceManager::start_all()
```

Comment:
`start_all` and `stop_all` are unclear, similarly to the title of this PR.
You aren't starting resources, you are starting all *components* and resources are acquired from them. I propose to rename these functions to `start_all_components` and `stop_all_components` or just `start_components` and `stop_components` but the danger with the latter is that someone may get the idea of introducing an argument list and then we are down a rabbit hole of what has been started and what not, better only have start all and stop all interfaces for the components.

Reply:
In the mid-term we will habe to have argument list and support for starting individual components.

The components are actually hardware, or hardware abstractions if you want to be exact.

For now I will rename it as proposed.

Reply:
Maybe when we are not winging it. I'd like to see a design draft and some strong arguments for it.

Reply:
Tool changers.

Reply:
I go with the "dangerous" version, so later we do not have to rename :)